mempool/src/shared_mempool.rs

```diff
@@ -448,18 +466,54 @@ where
     crit!("SharedMempool outbound_sync_task terminated");
 }
 
-async fn process_consensus_request<V>(
+async fn gc_committed_txns<V>(
```

Comment:
nit: commit_transactions instead. I'd prefer to use gc in mempool only for TTL-based garbage collection